Fix Project Panel `select_next_git_entry` action

Anthony-Eid opened the pull request:
## Context

I noticed that the project panel `select_next_git_entry` wasn't behaving correctly. Turns out it was searching in reverse, which caused the action to select itself or the last entry.

This PR corrects the behavior and adds a unit test that should stop regressions. 

Note: Since select next/prev git entry uses the same function as select next/prev diagnostic, the test partially works for that as well.

Release Notes:

- Fix bug where `select_next_git_entry` project panel action would only select a previous entry or the currently selected entry.

Anthony-Eid committed:
WIP initial test for select modified git entry

Anthony-Eid committed:
Fix project panel select next git entry

mikayla-maki commented on crates/project_panel/src/project_panel.rs:
```diff
@@ -6666,6 +6666,285 @@ mod tests {
+        )
+        .await;
+
+        fs.insert_tree(
```
nit: Zed style for these kinds of tests is to have all files and directories under the same `insert_tree` call. Let's combine these together.

Anthony-Eid replied:
I used two different calls to `insert_tree` because I wanted to test a case where a user has multiple worktrees in their project.

Is there a way to use `fs.insert_tree` to insert multiple worktrees I could use instead?

mikayla-maki replied:
You can just open those folders directly in the project: https://github.com/zed-industries/zed/blob/2b4a83918152f9695ddf7adfd47653d44e514146/crates/project/src/project_tests.rs#L1019

Anthony-Eid replied:
Thank you! Will push a change very soon

mikayla-maki commented on crates/project_panel/src/project_panel.rs:
```diff
@@ -6666,6 +6666,285 @@ mod tests {
+
+        let root1_dot_git = Path::new("/root1/.git");
+        let root2_dot_git = Path::new("/root2/.git");
+        let set_value = git::status::FileStatus::Tracked(git::status::TrackedStatus {
```
Let's switch these to imports rather than spelling out the whole path.

Anthony-Eid replied:
Will change

Anthony-Eid replied:
Done

mikayla-maki reviewed (changes requested):
Looks great! Just some small style nits :)

mikayla-maki commented on crates/project_panel/src/project_panel.rs:
```diff
@@ -6666,6 +6666,285 @@ mod tests {
+            panel.select_next_git_entry(&SelectNextGitEntry, window, cx);
+        });
+
+        assert_eq!(
```
What's going on with this assert? I looked at the ones above it and it seems to go `modified2` -> `modified4` -> `modified3`. Shouldn't it go in order?

Anthony-Eid replied:
Because `modified4` is in a directory it's placed ahead of `modified3` when project panel sorts the files. I agree that it makes more sense to have the order go 2-3-4 so I'll change the names to address this.

Anthony-Eid committed:
Address PR feedback from Mikayla

I switched to only using one fs.insert_tree call, changed the file names to
add clarity to the correct slection order, and added git::status::... as a
test function import to clean up the code

Anthony-Eid commented:
Please let me know if there's anymore feedback that I could use to improve the PR.

mikayla-maki committed:
Merge branch 'main' into fix-select-next

mikayla-maki commented:
Thank you!
